src/util.c

```diff
@@ -160,12 +160,12 @@ get_relative_date(const struct time *time, char *buf, size_t buflen, bool compac
 
 		seconds /= reldate[i].in_seconds;
 		if (compact) {
-			if (!string_nformat(buf, buflen, NULL, "%s%ld%c",
+			if (!string_nformat(buf, buflen, NULL, "%s%"PRIdMAX"%c",
 				    now.tv_sec >= timestamp ? "" : "-",
 				    seconds, reldate[i].compact_symbol))
 				return "";
 
-		} else if (!string_nformat(buf, buflen, NULL, "%ld %s%s %s",
+		} else if (!string_nformat(buf, buflen, NULL, "%"PRIdMAX" %s%s %s",
```

Comment:
If you use this format specifier, what you should do is cast the value as well (so `(intmax_t)value` iirc). Otherwise this will read 64 bits from a 32 bit field on 32bit glibc, for example.

Generally speaking, I'd recommend doing `%lld` for format and `(long long)value` instead.

This goes for all instances.

Reply:
Agreed, let's go the BSD way.

I actually took example on Git but I overlooked that they define their own timestamp_t type (though there was a clue as they use the unsigned version PRIuMAX).